tuxemon/states/combat/combat.py

```diff
@@ -1204,14 +1204,18 @@ def evolve(self) -> None:
                 # check the path field, path field signals evolution item based
                 if not evolution.path:
                     if evolution.at_level <= monster.level:
-                        logger.info(
-                            "{} evolved into {}!".format(
-                                monster.name, evolution.monster_slug
-                            )
-                        )
                         self.players[0].evolve_monster(
                             monster, evolution.monster_slug
                         )
+                elif evolution.path == "gender":
+                    if evolution.at_level <= monster.level:
+                        if evolution.gender == monster.gender:
+                            self.players[0].evolve_monster(
+                                monster, evolution.monster_slug
+                            )
+                elif evolution.path == "exception":
+                    if evolution.at_level <= monster.level:
```

Comment:
For exception evolutions it looks like we will evolve them multiple times... Maybe we should return after running this method?